backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/projection/impl/ElasticsearchSearchProjection.java

```diff
@@ -63,7 +63,7 @@ E extract(ProjectionHitMapper<?, ?> projectionHitMapper,
 	@SuppressWarnings({ "unchecked", "rawtypes" })
 	static <Z> Z transformUnsafe(ElasticsearchSearchProjection<?, Z> projection, LoadingResult<?> loadingResult,
 			Object extractedData, SearchProjectionTransformContext context) {
-		return (Z) ( (ElasticsearchSearchProjection) projection ).transform( loadingResult, extractedData, context );
```

Comment:
This cast is necessary to bypass generics limitations of the compiler. We can't remove it.